### DISCUSSION TITLE
Is there any intention of making Overlay/PopOver usable without Whisper?

### DISCUSSION THREAD

**Discussion by radix** [rsuite/rsuite#3061] [Q&A]
When I want to render a popover, but I have a regular old prop that I want to use to control whether the popover is visible, I have to jump through hoops to make Whisper work: use `trigger="none"`, create a ref for the Whisper, and then have a `useEffect` depending on my prop to call ref.open() or ref.close().

My first thought was to just use Popover by itself, which is close, but it doesn't give me any placement options, so I browsed the code some more and found out that the placement is handled by a component called `Overlay`, which `Whisper` wraps around the speaker (via `OverlayTrigger`).

Unfortunately, not only is `Overlay` undocumented, I couldn't *quite* figure out how to use it even with browsing the source; there's some context called `OverlayContext` that it seems to depend on and which applications can't import. I almost got this working by just calling `usePortal` with a context of `undefined`, but that doesn't seem to be the same behavior that I got with `Whisper`.

I would hope that I should just be able to write something like `<Overlay open={myProp}><Popover>...</Popover></Overlay>`.

**Comment by SevenOutman** [comment]
Looks like Whisper is needing a prop for controlled open state 🤔, something like

```jsx
<Whisper
  open={open}
  speaker={<Popover>...</Popover>}
>
  Content
</Whisper>
```

@radix Does it seem likely to help with your condition?

---
/cc @simonguo What do you think?

**Reply by radix** [reply]
@SevenOutman yes, that would be great.

**Reply by simonguo** [reply]
This feature can be added.
